src/macros.rs

```diff
@@ -138,6 +138,29 @@ macro_rules! define_from {
     }
 }
 
+macro_rules! define_from_smaller {
+    ($to:ident, $ty:ident, $($from:ident),+) => {
+        $(
+            impl From<$from> for $to {
+                #[inline(always)]
+                fn from(f: $from) -> $to {
+                    use std::mem::size_of;
+                    use std::ptr;
+
+                    let mut a = $to::splat(0 as $ty);
+                    unsafe {
+                        ptr::copy_nonoverlapping(
+                            &f as *const $from as *const u8,
+                            &mut a as *mut $to as *mut u8,
+                            size_of::<$from>());
```

Comment:
`size_of::<$from>()` should always be less than `size_of<$to>()`, but I still feel somewhat uneasy about this being unchecked.

Reply:
Same